kernel/kernel-util/src/main/java/org/sakaiproject/content/util/ZipContentUtil.java

```diff
@@ -82,6 +84,59 @@ public static int getMaxZipExtractFiles() {
 	public void compressFolder(Reference reference) {
 		compressFolder(reference, null, null);
 	}
+
+	public void compressSelectedResources(String siteId, String siteTitle, List<String> selectedFolderIds, List<String> selectedFiles, HttpServletResponse response) {
+		Map<String, ContentResource> resourcesToZip = new HashMap<>();
+
+		try {
+			// Add any files in the selected folders to the files to be in the zip.
+			if (selectedFolderIds.size() > 0) {
+				for (String selectedFolder : selectedFolderIds) {
+					List<ContentResource> folderContents = ContentHostingService.getAllResources(selectedFolder);
+					for (ContentResource folderFile : folderContents) {
+						resourcesToZip.put(folderFile.getId(), folderFile);
+					}
+				}
+			}
+
+			// Add any selected files to the list of resources to be in the zip.
+			for (String selectedFile : selectedFiles) {
+				ContentResource contentFile = ContentHostingService.getResource(selectedFile);
+				resourcesToZip.put(contentFile.getId(), contentFile);
+			}
+		} catch (IdUnusedException | PermissionException | TypeException e) {
+			// shouldn't happen by this stage.
+			LOG.error(e.getMessage(), e);
+		}
+
+		try (OutputStream zipOut = response.getOutputStream(); ZipOutputStream out = new ZipOutputStream(zipOut)) {
+			// If in dropbox need to add the word Dropbox to the end of the zip filename - use the first entry in the resourcesToZip map to find if we are in the dropthe user ID.
+			if (!resourcesToZip.isEmpty()) {
+				String firstContentResourceId = resourcesToZip.entrySet().iterator().next().getKey();
+				if (ContentHostingService.isInDropbox(firstContentResourceId) && ServerConfigurationService.getBoolean("dropbox.zip.haveDisplayname", true)) {
+					response.setHeader("Content-disposition", "inline; filename=" + siteId + "DropBox.zip");
```

Comment:
If you're supplying a filename then the disposition should be `attachment` I think.

Reply:
Ok thanks.